### PR TITLE
Do not filter out platform objects when doing dictionary conversions in overload resolution

### PR DESCRIPTION
**Pull request by Manishearth**
https://heycam.github.io/webidl/#es-overloads

In step 12, the platform object check is for substep 4, but importantly it only matters if `V` implements the matching interface. If not, it should be able to fall back to substep 10 and attempt conversion to a dictionary.

### CODE REVIEW

**Comment by Manishearth** [comment]
r? @jdm 

@bors-servo try


The specific thing triggering it was the following WebIDL and code:

```webidl
interface XRRay {
  constructor(optional DOMPointInit origin = {}, optional DOMPointInit direction = {});
  constructor(XRRigidTransform transform)
}
```

```js
new XRRay(new DOMPoint({x: 1, y: 0, z: 0}))
```


I can wait for XRRay to land to add a test for this, or I can add stuff to TestBinding. Thoughts?

**Comment by jdm** [comment]
Testbinding would be my preference.

**Comment by Manishearth** [comment]
Added a test

**Comment by jdm** [comment]
@bors-servo r-
Needs a `./mach fmt`.

**Comment by Manishearth** [comment]
Uh, what? the interface is enabled in the ini, the ini is pushed in this PR, and it works locally

File: tests/wpt/mozilla/meta/mozilla/overload_dict.html.ini [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[overload_dict.html]
+  type: testharness
+  prefs: [dom.testbinding.enabled:true]
```
Comment (jdm) [inline tests/wpt/mozilla/meta/mozilla/overload_dict.html.ini:3]: We need to duplicate this file in tests/wpt/mozilla/meta-layout-2020 as well.

**Comment by Manishearth** [comment]
@bors-servo retry

 - webdriver connection failures

**Comment by jdm** [comment]
@bors-servo retry
* #26161

**Pull request merged**